src/commands/moderation/clear.ts

```diff
@@ -0,0 +1,80 @@
+import { ApplicationCommandOptionType, ApplicationCommandType, TextChannel, Colors } from 'discord.js';
+import { slashCommand, slashCommandStructure, makeEmbed, constantsConfig, Logger } from '../../lib';
+
+const data = slashCommandStructure({
+    name: 'clear',
+    description: 'Delete a specified number of messages in the current channel.',
+    type: ApplicationCommandType.ChatInput,
+    default_member_permissions: constantsConfig.commandPermission.MANAGE_SERVER,
+    dm_permission: false,
+    options: [{
+        name: 'amount',
+        description: 'Number of messages to delete (1-100).',
+        type: ApplicationCommandOptionType.Integer,
+        required: true,
+        min_value: 1,
+        max_value: 100,
+    }],
+});
+
+export default slashCommand(data, async ({ interaction }) => {
+    const amount = interaction.options.getInteger('amount');
+
+    if (!amount) {
+        return interaction.reply({ content: 'Please specify the number of messages to delete.', ephemeral: true });
+    }
+
+    const { channel, user } = interaction;
+    if (!channel) {
+        return interaction.reply({ content: 'The channel could not be resolved.', ephemeral: true });
+    }
+
+    try {
+        const messages = await (channel as TextChannel).bulkDelete(amount, true);
+
+        const replyEmbed = makeEmbed({
+            title: 'Messages Deleted',
+            description: `Successfully deleted **${messages.size}** messages.`,
+            color: Colors.Green,
+            footer: { text: `Requested by ${user.tag}`, iconURL: user.displayAvatarURL() },
```

Comment:
This is an ephermal embed. Do we need the footer telling the moderator that they, themselves requested the messages to be deleted? Especially as no one else will see that `replyEmbed`?